### PR TITLE
support for Istanbul as a fork

### PR DESCRIPTION
**Pull request by shemnon**
* Adds support for Istanbul as a fork.
  * fork block
  * precompiels
  * block reqard
* Simpify the code branches to reduce the amount of code for future
  forks.

### CODE REVIEW

File: retesteth/ethObjects/object.cpp [hunk 1]
```diff
@@ -237,52 +237,39 @@ DataObject object::prepareGenesisParams(std::string const& _network, std::string
 {
     ClientConfig const& cfg = Options::get().getDynamicOptions().getCurrentConfig();
     test::checkAllowedNetwork(_network, cfg.getNetworks());
-    bool networkChecked = false;
```
Comment (winsvega) [inline retesteth/ethObjects/object.cpp]: this function I want to move to .json configs. it looks way to hardcode. 
the chain params config might be put in a .json in ini folder.

**Comment by winsvega** [comment]
added more configs for this params: https://github.com/ethereum/tests/pull/625

**Comment by winsvega** [comment]
@shemnon I miss one more function from testeth such as `--vmtrace`
this could be implemented as a debug method. it simply provide vm step by step trace. 
it is not required to generate/run the tests, but it really helps to write tests because you could trace
OOG exceptions and see if a contract in state test really get executed. 

do you have suggestions how to implement it in RPC ?

**Comment by shemnon** [comment]
Geth and parity have tracing RPC that may be useful for this, but they are quite verbose so the reference test would need some rich way to pick a value out of it.

Geth's tracing : https://github.com/ethereum/go-ethereum/wiki/Tracing:-Introduction
Parity's tracing : https://wiki.parity.io/JSONRPC-trace-module#trace_call

**Comment by shemnon** [comment]
Closed as the referenced commits cover this PR's goals quite nicely.

**Pull request closed without merging**